components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/MyAccountApplicationCreationListener.java

```diff
@@ -0,0 +1,194 @@
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.api.resource.mgt.APIResourceManager;
+import org.wso2.carbon.identity.api.resource.mgt.APIResourceMgtException;
+import org.wso2.carbon.identity.application.common.IdentityApplicationManagementException;
+import org.wso2.carbon.identity.application.common.model.APIResource;
+import org.wso2.carbon.identity.application.common.model.AuthorizedAPI;
+import org.wso2.carbon.identity.application.common.model.ServiceProvider;
+import org.wso2.carbon.identity.application.mgt.ApplicationManagementService;
+import org.wso2.carbon.identity.application.mgt.AuthorizedAPIManagementService;
+import org.wso2.carbon.identity.application.mgt.AuthorizedAPIManagementServiceImpl;
+import org.wso2.carbon.identity.application.mgt.internal.ApplicationManagementServiceComponentHolder;
+import org.wso2.carbon.identity.core.migrate.MigrationClientException;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.role.v2.mgt.core.RoleManagementService;
+import org.wso2.carbon.identity.role.v2.mgt.core.exception.IdentityRoleManagementException;
+import org.wso2.carbon.identity.role.v2.mgt.core.model.Permission;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wso2.carbon.identity.api.resource.mgt.constant.APIResourceManagementConstants.APIResourceTypes.TENANT;
+import static org.wso2.carbon.identity.api.resource.mgt.constant.APIResourceManagementConstants.RBAC_AUTHORIZATION;
+import static org.wso2.carbon.identity.application.mgt.ApplicationConstants.IMPERSONATE_ROLE_NAME;
+import static org.wso2.carbon.identity.application.mgt.ApplicationConstants.IMPERSONATE_SCOPE_NAME;
+import static org.wso2.carbon.identity.application.mgt.ApplicationConstants.MY_ACCOUNT_APPLICATION_NAME;
+import static org.wso2.carbon.identity.application.mgt.ApplicationConstants.USER_SESSION_IMPERSONATION_ENABLED;
+
+/**
+ * Listener to handle post tasks of My Account application creation.
+ */
+public class MyAccountApplicationCreationListener extends AbstractApplicationMgtListener {
+
+    private final AuthorizedAPIManagementService authorizedAPIManagementService = new
+            AuthorizedAPIManagementServiceImpl();
+
+    @Override
+    public int getExecutionOrderId() {
+
+        return 1;
+    }
+
+    @Override
+    public int getDefaultOrderId() {
+
+        return 1;
+    }
+
+    @Override
+    public boolean isEnable() {
+
+        return IdentityUtil.getProperty(USER_SESSION_IMPERSONATION_ENABLED) != null
+                && Boolean.parseBoolean(IdentityUtil.getProperty(USER_SESSION_IMPERSONATION_ENABLED));
+    }
+
+    @Override
+    public boolean doPostCreateApplication(ServiceProvider serviceProvider, String tenantDomain, String userName)
+            throws IdentityApplicationManagementException {
+
+        String appName = serviceProvider.getApplicationName();
+        if (MY_ACCOUNT_APPLICATION_NAME.equals(appName)) {
```

Comment:
should this be a true or false condition? Seems need to negate

Reply:
Fixed with 39ba690eb3dac3bb30f8f2b9979e583bd89950ab